pkg/machine/applehv/rest.go

```diff
@@ -94,9 +94,10 @@ func (vf *VfkitHelper) stop(force, wait bool) error {
 		if err := vf.stateChange(define.HardStop); err != nil {
 			return err
 		}
-	}
-	if err := vf.stateChange(define.Stop); err != nil {
-		return err
+	} else {
+		if err := vf.stateChange(define.Stop); err != nil {
+			return err
+		}
```

Comment:
It would probably be good, if possible, to add a test to check that the state is Stop or HardStop as appropriate.

Reply:
:+1:

Reply:
if i understand the comment, this test exists.